plugins/cpp/service/include/service/cppservice.h

```diff
@@ -164,6 +164,8 @@ class CppServiceHandler : virtual public LanguageServiceIf
 
     OVERRIDDEN_BY, /*!< This option returns the overrides of a function. */
 
+    VAR_TYPE, /*!< Type of a variable. */
```

Comment:
I created a new reference type (The TYPE keyword reserverd in the FileReferenceType that's the reason why I used VAR_TYPE) to query the type of a variable. We should overthink it.